lib/content-services/src/lib/search/components/search-filter-tabbed/search-filter-tabbed.component.ts

```diff
@@ -0,0 +1,80 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Component, ContentChildren, EventEmitter, Input, Output, QueryList, ViewEncapsulation } from '@angular/core';
+import { SearchFilterTabDirective } from './search-filter-tab.directive';
+import { SearchWidgetSettings } from '../../models/search-widget-settings.interface';
+import { TranslateService } from '@ngx-translate/core';
+
+@Component({
+  selector: 'adf-search-filter-tabbed',
+  templateUrl: './search-filter-tabbed.component.html',
+  styleUrls: ['./search-filter-tabbed.component.scss'],
+  encapsulation: ViewEncapsulation.None
+})
+export class SearchFilterTabbedComponent {
+    @Input()
+    set settings(settings: SearchWidgetSettings) {
+        this.fieldsChanged.emit(settings?.field.split(',').map(field => field.trim()));
+        this.displayedLabelsByField = settings.displayedLabelsByField ? Object.entries<string>(settings.displayedLabelsByField)
+            .reduce((displayLabelsByField, displayLabelAndField) => ({
+                [displayLabelAndField[0]]: this.translateService.instant(displayLabelAndField[1]),
+                ...displayLabelsByField
+            }), {}) : {};
+        this.displayedLabelsByFieldTranslated.emit(this.displayedLabelsByField);
+    }
+
+    @Input()
+    set queries(queries: { [key: string]: string }) {
+        this.queriesCombined.emit(Object.values(queries)
+            .reduce((wholeQuery, query) => wholeQuery ? `${wholeQuery} AND ${query}` : query, ''));
+    }
+
+    @Input()
+    set valuesToDisplay(valuesToDisplay: { [key: string]: string }) {
+        this.valuesToDisplayCombined.emit(Object.values(valuesToDisplay).every((value) => !value) ?
+            '' : Object.entries(valuesToDisplay).reduce((wholeValueToDisplay, valueToDisplayByField) => this.getWholeDisplayValue(wholeValueToDisplay, valueToDisplayByField), ''));
+    }
+
+    @Output()
+    fieldsChanged = new EventEmitter<string[]>();
+    @Output()
+    displayedLabelsByFieldTranslated = new EventEmitter<{ [key: string]: string }>();
+    @Output()
+    queriesCombined = new EventEmitter<string>();
+    @Output()
+    valuesToDisplayCombined = new EventEmitter<string>();
+
+    @ContentChildren(SearchFilterTabDirective)
+    tabsContents: QueryList<SearchFilterTabDirective>;
+
+    private displayedLabelsByField: { [key: string]: string };
+
+    constructor(private translateService: TranslateService) {}
+
+    private getWholeDisplayValue(wholeValueToDisplay, valueToDisplayByField): string {
+        let displayValue = '';
+        if(valueToDisplayByField && valueToDisplayByField[1] !== '') {
+            if(wholeValueToDisplay && wholeValueToDisplay !== '') {
+                displayValue = wholeValueToDisplay.concat(' ').concat(`${this.displayedLabelsByField[valueToDisplayByField[0]].toUpperCase()}: ${valueToDisplayByField[1]}`);
```

Comment:
Why are you using concat function to add strings instead just + or string templates?

Reply:
To me, concat() made it much more clearer that there is a blank space between the existing value and the new value we are appending. With string templates, the presence of a blank space is not that clear

``displayValue = `${wholeValueToDisplay} ${this.displayedLabelsByField[valueToDisplayByField[0]].toUpperCase()}: ${valueToDisplayByField[1]}`;``

Using + operator **only** would require using a lot of operators. 

`displayValue = wholeValueToDisplay + ' ' + this.displayedLabelsByField[valueToDisplayByField[0]].toUpperCase() + ': ' + valueToDisplayByField[1];`

We can use a mix of + and string templates, but that just seems like a weird combination - 

``displayValue = wholeValueToDisplay + ' ' + `${this.displayedLabelsByField[valueToDisplayByField[0]].toUpperCase()}: ${valueToDisplayByField[1]}`;``

Reply:
However, reading up on the concat() method, there seems to be performance impacts of it. So yes, we should use one of the alternatives listed above